share/completions/snap.fish

```diff
@@ -0,0 +1,280 @@
+# Completions for `snap` command
+
+function __fish_snap_no_subcommand --description 'Test if snap has yet to be given the subcommand'
+	for i in (commandline -opc)
+		if contains -- $i abort ack alias aliases buy changes connect disable disconnect download\
+            enable find get help info install interfaces known list login logout prefer refresh remove\
+            revert run set tasks try unalias version watch 
+			return 1
+		end
+	end
+	return 0
+end
+
+function __fish_snap_using_subcommand --description 'Test if given subcommand is used'
+    for i in (commandline -opc)
+        if contains -- $i $argv[1]
+            return 0
+        end
+    end
+    return 1
+end
+
+function __fish_snap_use_package --description 'Test if snap command should have packages as potential completion'
+	for i in (commandline -opc)
+		if contains -- $i alias buy disable download enable info install refresh remove revert run try
+			return 0
+		end
+	end
+	return 1
+end
+
+function __fish_snap_use_file --description 'Test if snap command should have files as potential completion'
+	for i in (commandline -opc)
+		if contains -- $i ack try
+			return 0
+		end
+	end
+	return 1
+end
+
+function __fish_snap_subcommand
+    set subcommand $argv[1]; set -e argv[1]
+    complete -f -c snap -n '__fish_snap_no_subcommand' -a $subcommand $argv
+end
+
+function __fish_snap_disabled_snaps --description 'List disabled snaps'
+    snap list | grep disabled | cut -d ' ' -f 1
```

Comment:
I'd suggest using the `string` builtin here, and throughout the completion where you use grep/cut etc.